--- oggwriter.go.org	2024-10-24 10:52:25
+++ oggwriter.go	2024-10-24 10:49:11
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: MIT
 
 // Package oggwriter implements OGG media container writer
-package oggwriter
+package suzu
 
 import (
 	"encoding/binary"
@@ -12,7 +12,6 @@
 
 	"github.com/pion/rtp"
 	"github.com/pion/rtp/codecs"
-	"github.com/pion/webrtc/v4/internal/util"
 )
 
 const (
@@ -68,7 +67,7 @@
 		stream:        out,
 		sampleRate:    sampleRate,
 		channelCount:  channelCount,
-		serial:        util.RandUint32(),
+		serial:        RandUint32(),
 		checksumTable: generateChecksumTable(),
 
 		// Timestamp and Granule MUST start from 1
